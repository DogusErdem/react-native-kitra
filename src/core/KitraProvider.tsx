import type { ComponentType, ForwardedRef } from 'react';
import React from 'react';
import { GestureHandlerRootView } from 'react-native-gesture-handler';
import Animated from 'react-native-reanimated';
import { StyleProp, ViewStyle } from 'react-native';

import { ThemeProvider } from './theme/theme';
import { TypographyContext, TypographyProvider } from './typography/typography';
import { NotificationContextType, NotificationProvider } from './notification/notification';

const Feather = require('react-native-vector-icons/Feather').default;
const Ionicons = require('react-native-vector-icons/Ionicons').default;
const AntDesign = require('react-native-vector-icons/AntDesign').default;

Animated.addWhitelistedNativeProps({ text: true });
const messageTypes = (theme:any) => ({
  SUCCESS: {
    backgroundColor: theme.colors.status.successLight,
    icon: <Feather name="check" size={25} color={theme.colors.status.success} />,
  },
  INFO: {
    backgroundColor: theme.colors.system.primary15,
    icon: <Ionicons name="information-circle-outline" size={25} color={theme.colors.system.primary} />,
  },
  DANGER: {
    backgroundColor: theme.colors.status.errorLight,
    icon: <AntDesign name="warning" size={25} color={theme.colors.status.error} />,
  },
  WARNING: {
    backgroundColor: theme.colors.status.warningLight,
    icon: <AntDesign name="warning" size={25} color={theme.colors.status.warning} />,
  },
});

type KitraProviderType= {
  children?:any,
  limit?:number,
  notificationRef?:React.Ref<NotificationContextType>,
  messageType?:(theme?:any, colorScheme?:'dark' | 'light') => {[key:string]: {
      backgroundColor:string,
    icon?:React.ReactNode
    onPress?:()=>void
  }},
  notificationCcontainerStyle?:StyleProp<ViewStyle>,
  customView?:({ header, type, message, theme }:
    {header:string, type:string, message:string, theme:any})=>React.ReactNode}

export const KitraProvider =
({ children, messageType = messageTypes, notificationCcontainerStyle, limit, customView,
  notificationRef }: KitraProviderType) => (
    <GestureHandlerRootView style={{ flex: 1 }}>
      <ThemeProvider>
        <TypographyProvider>
          <NotificationProvider
            messageType={theme => (messageType ? messageType(theme) : theme)}
            notificationCcontainerStyle={notificationCcontainerStyle}
            limit={limit}
            customView={item => customView?.(item)}
            ref={notificationRef}
          >
            {children}
          </NotificationProvider>
        </TypographyProvider>
      </ThemeProvider>
    </GestureHandlerRootView>
);

export function applyDefaults<T>(Component: ComponentType<T>) {
  return React.forwardRef((props: T, ref: ForwardedRef<any>) => (
    <TypographyContext.Consumer>
      {typography =>
        <Component ref={ref} typography={typography?.typography} {...props} />
          }
    </TypographyContext.Consumer>
  ));
}
