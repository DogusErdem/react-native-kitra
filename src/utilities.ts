/* eslint-disable no-bitwise */

import type React from 'react';
import type { IconType } from './types';
import { x11Colors } from './core/theme/x11';

function hexToRgbA(hex:string, percentage:number) {
  if (hex[0] !== '#') {
    hex = `#${x11ColorToHex(hex)}`;
  }

  let c;
  if (/^#([A-Fa-f0-9]{3}){1,2}$/.test(hex)) {
    c = hex.substring(1).split('');
    if (c.length === 3) {
      c = [c[0], c[0], c[1], c[1], c[2], c[2]];
    }
    c = `0x${c.join('')}`;
    // @ts-ignore
    return `rgba(${[(c >> 16) & 255, (c >> 8) & 255, c & 255].join(',')},${percentage / 100})`;
  }
  throw new Error('Bad Hex');
}

export const opacity = (color:string, percentage: number) => hexToRgbA(color, percentage);

export function x11ColorToHex(colorName: string) {
  const hex = x11Colors.find(x => x.name === colorName)?.hex;
  if (!hex) {
    throw new Error(`Unknown X11 color name: ${colorName}`);
  }
  return hex;
}

const customIcons: any = {};

export const registerCustomIconType = (id: string, customIcon: any) => {
  customIcons[id] = customIcon;
};

export const getIconType = (type: IconType): any => {
  switch (type) {
    case 'zocial':
      return require('react-native-vector-icons/Zocial').default;
    case 'octicon':
      return require('react-native-vector-icons/Octicons').default;
    case 'material':
      return require('react-native-vector-icons/MaterialIcons').default;
    case 'material-community':
      return require('react-native-vector-icons/MaterialCommunityIcons')
        .default;
    case 'ionicon':
      return require('react-native-vector-icons/Ionicons').default;
    case 'foundation':
      return require('react-native-vector-icons/Foundation').default;
    case 'evilicon':
      return require('react-native-vector-icons/EvilIcons').default;
    case 'entypo':
      return require('react-native-vector-icons/Entypo').default;
    case 'font-awesome':
      return require('react-native-vector-icons/FontAwesome').default;
    case 'font-awesome-5':
      return require('react-native-vector-icons/FontAwesome5').default;
    case 'simple-line-icon':
      return require('react-native-vector-icons/SimpleLineIcons').default;
    case 'feather':
      return require('react-native-vector-icons/Feather').default;
    case 'antdesign':
    case 'ant-design':
      return require('react-native-vector-icons/AntDesign').default;
    case 'fontisto':
      return require('react-native-vector-icons/Fontisto').default;
    default:
      if (Object.prototype.hasOwnProperty.call(customIcons, type)) {
        return customIcons[type];
      }
      return require('react-native-vector-icons/MaterialIcons').default;
  }
};

export const getIconProperties = (node: React.ReactElement) => {
  const { type, name, size, color } = node.props;
  return {
    type,
    name,
    size,
    color,
  };
};
