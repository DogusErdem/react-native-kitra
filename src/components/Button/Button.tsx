/* eslint-disable no-nested-ternary */
import React, { useState } from 'react';
import { Platform, Pressable, PressableProps, StyleSheet, Text, TextStyle, ViewStyle } from 'react-native';
import type { ButtonProps, FCCWD } from '../../types';
import { applyDefaults } from '../../core/KitraProvider';

export const Button: FCCWD<ButtonProps & PressableProps> = (
  { theme,
    typography,
    size = 'medium',
    disabled = false,
    label = '',
    statusStyle,
    iconPosition = 'left',
    style,
    textStyle,
    icon,
    ...props },
) => {
  const [isPressed, setIsPressed] = useState(false);
  const fontStyles = {
    large: typography?.body.medium,
    medium: typography?.body.smedium,
    small: typography?.body.xsmedium,
  };

  const viewStyles = {
    default: {
      container: {
        backgroundColor: statusStyle?.default?.container?.backgroundColor || theme?.primary,
      },
      text: {
        color: statusStyle?.default?.text?.color || theme?.white,
      },
    },

    focused: {
      container: {
        backgroundColor: statusStyle?.focused?.container?.backgroundColor || theme?.focused,
      },
      text: {
        color: statusStyle?.focused?.text?.color || theme?.white,
      },
    },

    disabled: {
      container: {
        backgroundColor: statusStyle?.disabled?.container?.backgroundColor || theme?.disabledLight,
      },
      text: {
        color: statusStyle?.disabled?.text?.color || theme?.grey,
      },
    },
  };

  return (
    <Pressable
      onPressIn={() => setIsPressed(true)}
      onPressOut={() => setIsPressed(false)}
      style={({ pressed }) =>
        [
          styles.container,

          { flexDirection: iconPosition === 'left' ? 'row' : 'row-reverse' },

          label ?
            { paddingVertical: 15,
              paddingHorizontal: 30 }
            :
            { padding: (fontStyles[size].lineHeight - fontStyles[size].fontSize) / 2 + 10 },

          (pressed ?
            viewStyles.focused.container
            :
            viewStyles.default.container),

          disabled ? viewStyles.disabled.container : null,

          style,
        ]}
      disabled={disabled}
      {...props}
    >

      {icon && React.cloneElement(icon, {
        size: icon.props.size || fontStyles[size].fontSize,
        color: (disabled ? viewStyles.disabled.text.color : (isPressed ? (icon.props.color || viewStyles.focused.text.color) : (icon.props.color || viewStyles.default.text.color))),
        style: [label.length ? (iconPosition === 'left' ? { marginRight: 10 } : { marginLeft: 10 }) : null, icon.props?.style],
      })}
      <Text testID="button_text" style={[disabled ? viewStyles.disabled.text : (isPressed ? viewStyles.focused.text : viewStyles.default.text), { fontWeight: '500' }, fontStyles[size], textStyle]}>
        {label}
      </Text>
    </Pressable>
  );
};

export default applyDefaults(Button);

const styles = StyleSheet.create({
  container: {
    width: '100%',
    borderRadius: 5,
    alignItems: 'center',
    justifyContent: 'center',

  },
});
