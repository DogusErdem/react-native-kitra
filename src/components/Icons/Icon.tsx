import { getIconType } from '../../utilities';
import { applyDefaults } from '../../core/KitraProvider';
import type { FCCWD, IconComponentProps, IconType } from '../../types';

type IconProps = {
    type: IconType,
} & IconComponentProps;

const Icon:FCCWD<IconProps> = ({ type, ...props }) => {
  const IconComponent = getIconType(type || 'ionicons');

  return (
    <IconComponent {...props} />
  );
};

export default applyDefaults(Icon);
