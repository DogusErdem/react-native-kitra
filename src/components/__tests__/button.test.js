import { fireEvent, render, waitFor, act } from '@testing-library/react-native';
import { useContext } from 'react';
import { KitraProvider } from '../../core/KitraProvider';
import Icon from '../Icons/Icon';
import Button from '../Button/Button';
import { opacity } from '../../utilities';

/* eslint-disable no-undef */

const renderWithContext = children => render(<KitraProvider>{children}</KitraProvider>);

describe('Button', () => {
  it('default render', () => {
    const tree = renderWithContext(
      <Button />,
    );
    expect(tree).toMatchSnapshot();
  });
  describe('render with different size', () => {
    it('render with small', () => {
      const { getByTestId } = renderWithContext(
        <Button
          label="Press Me!"
          right={() => <Icon testID="rightIcon" type="material-community" name="penguin" size={10} />}
          left={() => <Icon testID="leftIcon" type="material-community" name="penguin" />}
          size="small"
        />,
      );
      const label = getByTestId('buttonLabel');
      const pressable = getByTestId('pressable');
      expect(label).toHaveStyle({ fontSize: 10, lineHeight: 16, fontFamily: 'Poppins-Medium', fontWeight: '500' });
      expect(pressable).toHaveStyle({ paddingHorizontal: 20, paddingVertical: 10.5 });
    });
    it('render with medium', () => {
      const { getByTestId } = renderWithContext(
        <Button
          label="Press Me!"
          right={() => <Icon testID="rightIcon" type="material-community" name="penguin" size={10} />}
          left={() => <Icon testID="leftIcon" type="material-community" name="penguin" />}
          size="medium"
        />,
      );
      const label = getByTestId('buttonLabel');
      const pressable = getByTestId('pressable');
      expect(label).toHaveStyle({ fontSize: 12, lineHeight: 18, fontFamily: 'Poppins-Medium', fontWeight: '500' });
      expect(pressable).toHaveStyle({ paddingHorizontal: 24, paddingVertical: 12 });
    });
    it('render with large', () => {
      const { getByTestId } = renderWithContext(
        <Button
          label="Press Me!"
          right={() => <Icon testID="rightIcon" type="material-community" name="penguin" size={10} />}
          left={() => <Icon testID="leftIcon" type="material-community" name="penguin" />}
          size="large"
        />,
      );
      const label = getByTestId('buttonLabel');
      const pressable = getByTestId('pressable');
      expect(label).toHaveStyle({ fontSize: 14, lineHeight: 21, fontFamily: 'Poppins-Medium', fontWeight: '500' });
      expect(pressable).toHaveStyle({ paddingHorizontal: 30, paddingVertical: 15 });
    });
  });
  describe('render with press', () => {
    it('render with onPressIn', async () => {
      const onPressInMock = jest.fn();
      const onPress = jest.fn();
      const { getByTestId } = renderWithContext(
        <Button
          label="Press Me!"
          onPress={onPress}
          onPressIn={onPressInMock}
        />,
      );
      const pressable = getByTestId('pressable');
      fireEvent(pressable, 'onPressIn');
      expect(onPressInMock).toHaveBeenCalledTimes(1);
    });
    it('render with onPressOut', async () => {
      const onPressOut = jest.fn();
      const onPress = jest.fn();
      const { getByTestId } = renderWithContext(
        <Button
          label="Press Me!"
          onPress={onPress}
          onPressOut={onPressOut}
        />,
      );
      const pressable = getByTestId('pressable');
      fireEvent(pressable, 'onPressOut');
      expect(onPressOut).toHaveBeenCalledTimes(1);
    });
  });
  describe('disabled button', () => {
    const onPress = jest.fn();
    it('render with disabled prop', () => {
      const { getByTestId } = renderWithContext(
        <Button
          label="Press Me!"
          disabled
          onPress={onPress}
        />,
      );
      const pressable = getByTestId('pressable');
      fireEvent(pressable, 'onPress');
      expect(onPress).toHaveBeenCalledTimes(0);
    });
  });
  describe('render with custom theme', () => {
    it('default theme', () => {
      const { getByTestId } = renderWithContext(
        <Button
          label="Press Me!"
          theme={{ default: { background: 'blue', label: 'white' } }}
        />,
      );
      const pressable = getByTestId('pressable');
      const buttonLabel = getByTestId('buttonLabel');
      expect(pressable).toHaveStyle({ backgroundColor: 'blue' });
      expect(buttonLabel).toHaveStyle({ color: 'white' });
    });
    it('disabled theme', () => {
      const { getByTestId } = renderWithContext(
        <Button
          label="Press Me!"
          theme={{ disabled: { background: 'grey', label: 'white' } }}
          disabled
        />,
      );
      const pressable = getByTestId('pressable');
      const buttonLabel = getByTestId('buttonLabel');
      expect(pressable).toHaveStyle({ backgroundColor: 'grey' });
      expect(buttonLabel).toHaveStyle({ color: 'white' });
    });
    // it('pressed theme', () => {
    //   const onPressInMock = jest.fn();
    //   const { getByTestId } = renderWithContext(
    //     <Button
    //       label="Press Me!"
    //       theme={{ pressed: { background: 'black', label: 'white' } }}
    //       onPressIn={onPressInMock}
    //     />,
    //   );
    //   const pressable = getByTestId('pressable');
    //   const buttonLabel = getByTestId('buttonLabel');
    //   fireEvent(pressable);
    //   expect(pressable).toHaveStyle({ backgroundColor: 'black' });
    //   expect(buttonLabel).toHaveStyle({ color: 'white' });
    // });
  });
  describe('render with custom styles', () => {
    it('render with labelStyle and style props', () => {
      const { getByTestId } = renderWithContext(
        <Button
          labelStyle={{ fontSize: 20, fontWeight: 'bold', lineHeight: 20 }}
          label="Test"
          style={{ padding: 10, marginTop: 10, paddingHorizontal: 10 }}
        />,
      );
      const pressable = getByTestId('pressable');
      const label = getByTestId('buttonLabel');

      expect(pressable).toHaveStyle({ padding: 10, marginTop: 10, paddingHorizontal: 10 });
      expect(label).toHaveStyle({ fontSize: 20, fontWeight: 'bold', lineHeight: 20 });
    });
  });
});
