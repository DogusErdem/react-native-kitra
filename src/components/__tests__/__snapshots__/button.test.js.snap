// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders small size heart icon to different positions renders small size heart icon to left 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityState={
      {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 5,
          "justifyContent": "center",
          "width": "100%",
        },
        {
          "flexDirection": "row",
        },
        {
          "paddingHorizontal": 30,
          "paddingVertical": 15,
        },
        {
          "backgroundColor": "#8973CD",
        },
        null,
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
          },
          {
            "fontWeight": "500",
          },
          {
            "fontFamily": "Poppins-Medium",
            "fontSize": 10,
            "lineHeight": 16,
          },
          undefined,
        ]
      }
      testID="button_text"
    >
      Button Text
    </Text>
  </View>
  <View
    entering={BaseAnimationMock {}}
    exiting={BaseAnimationMock {}}
    layout={BaseAnimationMock {}}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 10,
          "height": 100,
          "paddingHorizontal": 24,
          "width": "100%",
          "zIndex": 100,
        },
        {
          "alignSelf": "center",
          "top": 150,
        },
        undefined,
        {
          "backgroundColor": "transparent",
          "marginTop": -110,
          "position": "absolute",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flex": 1,
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "borderRadius": 100,
            "height": 28,
            "justifyContent": "center",
            "marginLeft": 18,
            "marginTop": 21,
            "width": 28,
            "zIndex": 100,
          }
        }
      />
      <View
        style={
          {
            "flex": 1,
            "zIndex": 100,
          }
        }
      >
        <Text
          ellipsizeMode="middle"
          numberOfLines={3}
          style={
            [
              {
                "fontWeight": "500",
                "marginTop": 17,
                "paddingHorizontal": 20,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        />
        <Text
          ellipsizeMode="middle"
          numberOfLines={3}
          style={
            [
              {
                "fontWeight": "400",
                "marginTop": 6,
                "paddingHorizontal": 20,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Button renders small size heart icon to different positions renders small size heart icon to right 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityState={
      {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 5,
          "justifyContent": "center",
          "width": "100%",
        },
        {
          "flexDirection": "row-reverse",
        },
        {
          "paddingHorizontal": 30,
          "paddingVertical": 15,
        },
        {
          "backgroundColor": "#8973CD",
        },
        null,
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
          },
          {
            "fontWeight": "500",
          },
          {
            "fontFamily": "Poppins-Medium",
            "fontSize": 10,
            "lineHeight": 16,
          },
          undefined,
        ]
      }
      testID="button_text"
    >
      Button Text
    </Text>
  </View>
  <View
    entering={BaseAnimationMock {}}
    exiting={BaseAnimationMock {}}
    layout={BaseAnimationMock {}}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 10,
          "height": 100,
          "paddingHorizontal": 24,
          "width": "100%",
          "zIndex": 100,
        },
        {
          "alignSelf": "center",
          "top": 150,
        },
        undefined,
        {
          "backgroundColor": "transparent",
          "marginTop": -110,
          "position": "absolute",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flex": 1,
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "borderRadius": 100,
            "height": 28,
            "justifyContent": "center",
            "marginLeft": 18,
            "marginTop": 21,
            "width": 28,
            "zIndex": 100,
          }
        }
      />
      <View
        style={
          {
            "flex": 1,
            "zIndex": 100,
          }
        }
      >
        <Text
          ellipsizeMode="middle"
          numberOfLines={3}
          style={
            [
              {
                "fontWeight": "500",
                "marginTop": 17,
                "paddingHorizontal": 20,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        />
        <Text
          ellipsizeMode="middle"
          numberOfLines={3}
          style={
            [
              {
                "fontWeight": "400",
                "marginTop": 6,
                "paddingHorizontal": 20,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Button renders text button with different sizes renders text button  1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityState={
      {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 5,
          "justifyContent": "center",
          "width": "100%",
        },
        {
          "flexDirection": "row",
        },
        {
          "paddingHorizontal": 30,
          "paddingVertical": 15,
        },
        {
          "backgroundColor": "#8973CD",
        },
        null,
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
          },
          {
            "fontWeight": "500",
          },
          {
            "fontFamily": "Poppins-Medium",
            "fontSize": 12,
            "lineHeight": 18,
          },
          undefined,
        ]
      }
      testID="button_text"
    >
      Button Text
    </Text>
  </View>
  <View
    entering={BaseAnimationMock {}}
    exiting={BaseAnimationMock {}}
    layout={BaseAnimationMock {}}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 10,
          "height": 100,
          "paddingHorizontal": 24,
          "width": "100%",
          "zIndex": 100,
        },
        {
          "alignSelf": "center",
          "top": 150,
        },
        undefined,
        {
          "backgroundColor": "transparent",
          "marginTop": -110,
          "position": "absolute",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flex": 1,
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "borderRadius": 100,
            "height": 28,
            "justifyContent": "center",
            "marginLeft": 18,
            "marginTop": 21,
            "width": 28,
            "zIndex": 100,
          }
        }
      />
      <View
        style={
          {
            "flex": 1,
            "zIndex": 100,
          }
        }
      >
        <Text
          ellipsizeMode="middle"
          numberOfLines={3}
          style={
            [
              {
                "fontWeight": "500",
                "marginTop": 17,
                "paddingHorizontal": 20,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        />
        <Text
          ellipsizeMode="middle"
          numberOfLines={3}
          style={
            [
              {
                "fontWeight": "400",
                "marginTop": 6,
                "paddingHorizontal": 20,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Button renders text button with different sizes renders text button with large size  1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityState={
      {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 5,
          "justifyContent": "center",
          "width": "100%",
        },
        {
          "flexDirection": "row",
        },
        {
          "paddingHorizontal": 30,
          "paddingVertical": 15,
        },
        {
          "backgroundColor": "#8973CD",
        },
        null,
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
          },
          {
            "fontWeight": "500",
          },
          {
            "fontFamily": "Poppins-Medium",
            "fontSize": 14,
            "lineHeight": 21,
          },
          undefined,
        ]
      }
      testID="button_text"
    >
      Button Text
    </Text>
  </View>
  <View
    entering={BaseAnimationMock {}}
    exiting={BaseAnimationMock {}}
    layout={BaseAnimationMock {}}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 10,
          "height": 100,
          "paddingHorizontal": 24,
          "width": "100%",
          "zIndex": 100,
        },
        {
          "alignSelf": "center",
          "top": 150,
        },
        undefined,
        {
          "backgroundColor": "transparent",
          "marginTop": -110,
          "position": "absolute",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flex": 1,
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "borderRadius": 100,
            "height": 28,
            "justifyContent": "center",
            "marginLeft": 18,
            "marginTop": 21,
            "width": 28,
            "zIndex": 100,
          }
        }
      />
      <View
        style={
          {
            "flex": 1,
            "zIndex": 100,
          }
        }
      >
        <Text
          ellipsizeMode="middle"
          numberOfLines={3}
          style={
            [
              {
                "fontWeight": "500",
                "marginTop": 17,
                "paddingHorizontal": 20,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        />
        <Text
          ellipsizeMode="middle"
          numberOfLines={3}
          style={
            [
              {
                "fontWeight": "400",
                "marginTop": 6,
                "paddingHorizontal": 20,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Button renders text button with different sizes renders text button with medium size  1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityState={
      {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 5,
          "justifyContent": "center",
          "width": "100%",
        },
        {
          "flexDirection": "row",
        },
        {
          "paddingHorizontal": 30,
          "paddingVertical": 15,
        },
        {
          "backgroundColor": "#8973CD",
        },
        null,
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
          },
          {
            "fontWeight": "500",
          },
          {
            "fontFamily": "Poppins-Medium",
            "fontSize": 12,
            "lineHeight": 18,
          },
          undefined,
        ]
      }
      testID="button_text"
    >
      Button Text
    </Text>
  </View>
  <View
    entering={BaseAnimationMock {}}
    exiting={BaseAnimationMock {}}
    layout={BaseAnimationMock {}}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 10,
          "height": 100,
          "paddingHorizontal": 24,
          "width": "100%",
          "zIndex": 100,
        },
        {
          "alignSelf": "center",
          "top": 150,
        },
        undefined,
        {
          "backgroundColor": "transparent",
          "marginTop": -110,
          "position": "absolute",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flex": 1,
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "borderRadius": 100,
            "height": 28,
            "justifyContent": "center",
            "marginLeft": 18,
            "marginTop": 21,
            "width": 28,
            "zIndex": 100,
          }
        }
      />
      <View
        style={
          {
            "flex": 1,
            "zIndex": 100,
          }
        }
      >
        <Text
          ellipsizeMode="middle"
          numberOfLines={3}
          style={
            [
              {
                "fontWeight": "500",
                "marginTop": 17,
                "paddingHorizontal": 20,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        />
        <Text
          ellipsizeMode="middle"
          numberOfLines={3}
          style={
            [
              {
                "fontWeight": "400",
                "marginTop": 6,
                "paddingHorizontal": 20,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Button renders text button with different sizes renders text button with small size  1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityState={
      {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 5,
          "justifyContent": "center",
          "width": "100%",
        },
        {
          "flexDirection": "row",
        },
        {
          "paddingHorizontal": 30,
          "paddingVertical": 15,
        },
        {
          "backgroundColor": "#8973CD",
        },
        null,
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
          },
          {
            "fontWeight": "500",
          },
          {
            "fontFamily": "Poppins-Medium",
            "fontSize": 10,
            "lineHeight": 16,
          },
          undefined,
        ]
      }
      testID="button_text"
    >
      Button Text
    </Text>
  </View>
  <View
    entering={BaseAnimationMock {}}
    exiting={BaseAnimationMock {}}
    layout={BaseAnimationMock {}}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 10,
          "height": 100,
          "paddingHorizontal": 24,
          "width": "100%",
          "zIndex": 100,
        },
        {
          "alignSelf": "center",
          "top": 150,
        },
        undefined,
        {
          "backgroundColor": "transparent",
          "marginTop": -110,
          "position": "absolute",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flex": 1,
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "borderRadius": 100,
            "height": 28,
            "justifyContent": "center",
            "marginLeft": 18,
            "marginTop": 21,
            "width": 28,
            "zIndex": 100,
          }
        }
      />
      <View
        style={
          {
            "flex": 1,
            "zIndex": 100,
          }
        }
      >
        <Text
          ellipsizeMode="middle"
          numberOfLines={3}
          style={
            [
              {
                "fontWeight": "500",
                "marginTop": 17,
                "paddingHorizontal": 20,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        />
        <Text
          ellipsizeMode="middle"
          numberOfLines={3}
          style={
            [
              {
                "fontWeight": "400",
                "marginTop": 6,
                "paddingHorizontal": 20,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
